### PR TITLE
Moved Chemistry system into the Systems Namespace

### PR DESCRIPTION
**Pull request by TaeruAlethea**
### Purpose
Name space organization is important to prevent future technical debt.
Moves the Chemistry system into the Systems Namespace.

### CODE REVIEW

**Comment by corp-0** [comment]
cool, although looks like this removed the chemistry asmdef which isn't desirable (we need to move stuff to their own definitions whenever possible to reduce compilation time when making changes)

**Comment by TaeruAlethea** [comment]
> cool, although looks like this removed the chemistry asmdef which isn't desirable (we need to move stuff to their own definitions whenever possible to reduce compilation time when making changes)

Chemistry having its own asmdef was a function of it being in its own namespace, to my knowledge.

**Comment by TaeruAlethea** [comment]
oops seems I broke this PR. I was trying to rename the branch in my fork to something more descriptive. Ill make sure to name stuff before submitting PRs

**Comment by corp-0** [comment]
> Chemistry having its own asmdef was a function of it being in its own namespace, to my knowledge.

No, it was deribately written like that, having isolation in mind. Chemistry doesn't depend on anything so whatever you change outside of the Chemsitry asmdef doesn't make Chemistry recompile. It is intended to move as much stuff as we can to their own asmdef to reduce compiling times.

**Comment by TaeruAlethea** [comment]
> > Chemistry having its own asmdef was a function of it being in its own namespace, to my knowledge.
> 
> No, it was deribately written like that, having isolation in mind. Chemistry doesn't depend on anything so whatever you change outside of the Chemsitry asmdef doesn't make Chemistry recompile. It is intended to move as much stuff as we can to their own asmdef to reduce compiling times.

I understand now. as soon as I can get the Unity Editor to let me do anything, I will recreate the Assembly definition.

**Comment by TaeruAlethea** [comment]
@corp-0, So after creating the Assembly Definition for Chemistry, Many things start to complain about the Namespace issues. Everything in the ./Chemistry/Effects/ directory is dependent on Systems namespace stuff.

**Comment by corp-0** [comment]
that's what I was thinking, so we have a problem right there because we want the systems to be under System namespace but also hopefully isolate them

**Comment by TaeruAlethea** [comment]
@corp-0 I dont think it was ever as isolated as it seemed. It was just using its position as a top level namespace to grow tendrils into other things.

**Comment by corp-0** [comment]
chemistry uses nothing, the rest of the game uses chemistry

**Comment by TaeruAlethea** [comment]
> chemistry uses nothing, the rest of the game uses chemistry

Chemisty/Effect has dependencies.

**Comment by NoooneyDude** [comment]
> Chemisty/Effect has dependencies.

I suppose Chemistry/Effect mightn't have been part of the original asmdef. I think there might've been others too.

File: UnityProject/Chemistry/Reagent.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Chemistry
+{
+	public class Reagent
+	{
+		public string Name { get; internal set; }
+	}
+}
```
Comment (NoooneyDude) [inline UnityProject/Chemistry/Reagent.cs]: What's the story with this one? Chilling in `UnityProject/Chemistry`?

Comment (TaeruAlethea) [reply UnityProject/Chemistry/Reagent.cs]: I'm not sure. It was in Chemistry befor, as far as I know. There is another "reagent" in Botany too.

Comment (TaeruAlethea) [reply UnityProject/Chemistry/Reagent.cs]: From what I can see, this version is not referenced by anything. I have removed it.

**Comment by TaeruAlethea** [comment]
Rebase to Unitystation/develop performed.

**Comment by TaeruAlethea** [comment]
The 3 errors under Assets/Scripts/Objects/Kitchen/InteractableGrinder.cs which was not changed by my changes.

**Comment by corp-0** [comment]
I can pull this and see what's wrong tomorrow

**Comment by corp-0** [comment]
can you please reset back to the last commit you did before rebasing so I can try rebasing on my local?

**Comment by TaeruAlethea** [comment]
Rebased again per @NoooneyDude 's request in the discord.

**Comment by corp-0** [comment]
I can solve the conflict for you, I just need this branch to be back to what it was before the attempt of solving conflicts

**Pull request closed without merging**